Panic upon self-shutdown

I don't think panicking in this case is necessary. :stuck_out_tongue: 

```
time="2020-02-17T13:39:30Z" level=panic msg="Could not connect to SQL for over 5 minutes. Shutting down..."
panic: (*logrus.Entry) (0xa5f0e0,0xc00026ad90)

goroutine 62 [running]:
github.com/sirupsen/logrus.Entry.log(0xc00009c620, 0xc000087c50, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:239 +0x2db
github.com/sirupsen/logrus.(*Entry).Log(0xc00026acb0, 0xc000000000, 0xc000311c78, 0x1, 0x1)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/entry.go:268 +0xeb
github.com/sirupsen/logrus.(*Logger).Log(0xc00009c620, 0x0, 0xc000311c78, 0x1, 0x1)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:192 +0x7d
github.com/sirupsen/logrus.(*Logger).Panic(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/logger.go:233
github.com/sirupsen/logrus.Panic(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.4.2/exported.go:129
github.com/Icinga/icingadb/connection.(*DBWrapper).getConnectionCheckInterval(0xc00017eba0, 0xc000311d20)
	/go/src/github.com/Icinga/icingadb/connection/mysql.go:141 +0xbf
github.com/Icinga/icingadb/connection.(*DBWrapper).checkConnection(0xc00017eba0, 0xdf8475801, 0xc000186500)
	/go/src/github.com/Icinga/icingadb/connection/mysql.go:162 +0x20f
github.com/Icinga/icingadb/connection.NewDBWrapper.func1(0xc00017eba0)
	/go/src/github.com/Icinga/icingadb/connection/mysql.go:101 +0x30
created by github.com/Icinga/icingadb/connection.NewDBWrapper
	/go/src/github.com/Icinga/icingadb/connection/mysql.go:99 +0x223
exit status 2
```

--- a/connection/mysql.go
+++ b/connection/mysql.go
@@ -138,7 +138,7 @@ func (dbw *DBWrapper) getConnectionCheckInterval() time.Duration {
 		} else if v < 14 {
 			return 60 * time.Second
 		} else {
-			log.Panic("Could not connect to SQL for over 5 minutes. Shutting down...")
+			log.Fatal("Could not connect to SQL for over 5 minutes. Shutting down...")
 		}
 	}
 

--- a/connection/mysql_test.go
+++ b/connection/mysql_test.go
@@ -10,6 +10,7 @@ import (
 	"github.com/Icinga/icingadb/config/testbackends"
 	"github.com/Icinga/icingadb/utils"
 	"github.com/go-sql-driver/mysql"
+	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
@@ -292,10 +293,18 @@ func TestGetConnectionCheckInterval(t *testing.T) {
 	atomic.StoreUint32(dbw.ConnectionLostCounterAtomic, 11)
 	assert.Equal(t, 60*time.Second, dbw.getConnectionCheckInterval())
 
-	//Should panic, if not connected and counter > 13
+	//Should exit, if not connected and counter > 13
 	dbw.CompareAndSetConnected(false)
 	atomic.StoreUint32(dbw.ConnectionLostCounterAtomic, 14)
-	assert.Panics(t, func() { dbw.getConnectionCheckInterval() }, "Should panic")
+
+	exited := false
+	defer func() { logrus.StandardLogger().ExitFunc = nil }()
+	logrus.StandardLogger().ExitFunc = func(i int) {
+		exited = true
+	}
+
+	dbw.getConnectionCheckInterval()
+	assert.Equal(t, true, exited, "Should have exited")
 }
 
 func TestDBWrapper_SqlFetchAll(t *testing.T) {

--- a/connection/redis.go
+++ b/connection/redis.go
@@ -168,7 +168,7 @@ func (rdbw *RDBWrapper) getConnectionCheckInterval() time.Duration {
 		} else if v < 14 {
 			return 60 * time.Second
 		} else {
-			log.Panic("Could not connect to Redis for over 5 minutes. Shutting down...")
+			log.Fatal("Could not connect to Redis for over 5 minutes. Shutting down...")
 		}
 	}
 

--- a/connection/redis_test.go
+++ b/connection/redis_test.go
@@ -5,6 +5,7 @@ package connection
 import (
 	"github.com/Icinga/icingadb/config/testbackends"
 	"github.com/go-redis/redis"
+	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"sync"
 	"sync/atomic"
@@ -54,10 +55,18 @@ func TestRDBWrapper_GetConnectionCheckInterval(t *testing.T) {
 	atomic.StoreUint32(rdbw.ConnectionLostCounterAtomic, 11)
 	assert.Equal(t, 60*time.Second, rdbw.getConnectionCheckInterval())
 
-	//Should panic, if not connected and counter > 13
+	//Should exit, if not connected and counter > 13
 	rdbw.CompareAndSetConnected(false)
 	atomic.StoreUint32(rdbw.ConnectionLostCounterAtomic, 14)
-	assert.Panics(t, func() { rdbw.getConnectionCheckInterval() }, "Should panic")
+
+	exited := false
+	defer func() { logrus.StandardLogger().ExitFunc = nil }()
+	logrus.StandardLogger().ExitFunc = func(i int) {
+		exited = true
+	}
+
+	rdbw.getConnectionCheckInterval()
+	assert.Equal(t, true, exited, "Should have exited")
 }
 
 func TestRDBWrapper_CheckConnection(t *testing.T) {

